src/foam/nanos/boot/NSpec.js

```diff
@@ -217,28 +270,68 @@ foam.CLASS({
           return service;
         }
 
-        Interpreter shell = new Interpreter();
-        try {
-          shell.set("x", x);
-          Object service = shell.eval(getServiceScript());
-          saveService(x, service);
-          return service;
-        } catch (EvalError e) {
-          foam.nanos.logger.Logger logger = (foam.nanos.logger.Logger) x.get("logger");
-          if ( logger != null ) {
-            logger.error("NSpec serviceScript error", getServiceScript(), e);
-          } else {
-            System.err.println("NSpec serviceScript error: " + getServiceScript());
-            e.printStackTrace();
+        Language l = getLanguage();
```

Comment:
I would prefer this not be in NSpec. Instead, create a JShellExecutor service in the context. One for BeanShell as well would be good, then just pass the script to that service for execution.